### DISCUSSION TITLE
Interoperate with `MonadState`

### DISCUSSION THREAD

**Discussion by drlkf** [haskell-effectful/effectful#162] [Q&A] [answered]
Following the documentation, which says:

> Therefore, what we do instead is provide an orphan, canonical instance of MonadRNG for Eff that delegates to the RNG effect

which I then try to apply to a `MonadState` instance:
```haskell
instance State s :> es => MonadState s (Eff es) where
  get = send Get
  put = send Put
```
which results in:
```
    • Illegal instance declaration for ‘MonadState s (Eff es)’
        The liberal coverage condition fails in class ‘MonadState’
          for functional dependency: ‘m -> s’
        Reason: lhs type ‘Eff es’ does not determine rhs type ‘s’
        Un-determined variable: s
    • In the instance declaration for ‘MonadState s (Eff es)’
```
I'm pretty sure I missed something, this case seems so basic that I must have overlooked something important, but I can't figure out what. Would you have any pointers?

**Comment by ocharles** [comment, marked as answer]
Try 

```haskell
instance (State s :> es, MonadState s (Eff s)) => MonadState s (Eff es) where
  get = send Get
  put = send Put
```

**Comment by arybczak** [comment]
@ocharles is right, including the instance head in the context allows you to bypass the coverage condition.

It's somewhat of a hack, but gets the job done. This is needed for any class that has fundeps.
